kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotHint.java

```diff
@@ -18,17 +18,35 @@
 
 import io.delta.kernel.internal.actions.Metadata;
 import io.delta.kernel.internal.actions.Protocol;
+import java.util.OptionalLong;
 
 /** Contains summary information of a {@link io.delta.kernel.Snapshot}. */
 public class SnapshotHint {
   private final long version;
   private final Protocol protocol;
   private final Metadata metadata;
+  private final OptionalLong tableSizeBytes;
+  private final OptionalLong numFiles;
 
   public SnapshotHint(long version, Protocol protocol, Metadata metadata) {
     this.version = version;
     this.protocol = protocol;
     this.metadata = metadata;
+    this.tableSizeBytes = OptionalLong.empty();
+    this.numFiles = OptionalLong.empty();
+  }
+
+  public SnapshotHint(
+      long version,
+      Protocol protocol,
+      Metadata metadata,
+      OptionalLong tableSizeBytes,
+      OptionalLong numFiles) {
+    this.version = version;
+    this.protocol = protocol;
+    this.metadata = metadata;
+    this.tableSizeBytes = tableSizeBytes;
+    this.numFiles = numFiles;
```

Comment:
Is `metadata` and `protocol` required non-null here? let's add `requireNonNull` to all these non-primitive args

Reply:
otherwise we need null checks up in `ChecksumWriter. buildCheckSumRow`